Fix flaky unit tests

```
--- FAIL: TestMainTestSuite/TestNomadRunnerManager_Load (2.30s)
--- FAIL: TestMainTestSuite/TestNomadRunnerManager_checkPrewarmingPoolAlert_reloadsRunners (1.20s)
```

- Failed: https://github.com/openHPI/poseidon/actions/runs/10979412090/job/30483777746
- Successful: https://github.com/openHPI/poseidon/actions/runs/10979412090/job/30483822611

--- a/internal/runner/nomad_manager_test.go
+++ b/internal/runner/nomad_manager_test.go
@@ -656,6 +656,7 @@ func (s *MainTestSuite) TestNomadRunnerManager_Load() {
 		}
 
 		cancelExecution()
+		<-time.After(tests.ShortTimeout)
 		err = r.Destroy(ErrLocalDestruction)
 		s.Require().NoError(err)
 	})

